module/VuFind/src/VuFind/Search/Factory/AbstractSolrBackendFactory.php

```diff
@@ -249,9 +250,8 @@ protected function getMergedIndexConfig(): array
     protected function getFlatIndexConfig(): array
     {
         $configList = $this->getPrioritizedConfigsForIndexSettings();
-        $configObj = $this->config->get($configList[0]);
-        return isset($configObj->Index)
-            ? $configObj->Index->toArray() : [];
+        $configObj = $this->configManager->getConfigArray($configList[0]);
+        return $configObj['Index'] ?? [];
```

Comment:
Suggested variable name improvement:

```suggestion
        $configArray = $this->configManager->getConfigArray($configList[0]);
        return $configArray['Index'] ?? [];
```